packages/clay-shared/src/useFocusManagement.ts

```diff
@@ -0,0 +1,224 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import {useEffect, useRef} from 'react';
+
+type FocusManager = {
```

Comment:
Why use a `type` vs an `interface`? Mostly asking because I have a hard time knowing what to use sometimes.

Reply:
Oh yeah, this can lead to confusion I'm also not completely sure about this, but I see that it makes sense to use:

- `type` for cases where it is not necessary a complete definition of something or that also do not need to use extends and functionalities that the `interface` allows me.
- `interface` I see more common usage for our `IProps` and when I need to extend other interfaces or when I create signatures for `class`, for example.

But I'm not 100% sure about this, sometimes I'm still in doubt, so I'm not sure when to use it or not in some cases. But what I'm trying to do now, I usually always use `interface`.

Reply:
I usually default to `type`, but TSLint [alleges that](https://palantir.github.io/tslint/rules/interface-over-type-literal/):

> Interfaces are generally preferred over type literals because interfaces can be implemented, extended and merged.

typescript-eslint had a similar rule, [prefer-interface](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/prefer-interface.md), but that one got deprecated in favor of the apparently less opinionated [consistent-type-definitions](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/consistent-type-definitions.md). That one enables you to always prefer `type`, or always prefer `interface`, and the only guidance it gives you is:

> If you specifically want to use an interface or type literal for stylistic reasons, you can disable this rule.

I think there are two reasonable positions on this one:

- Use `type` for everything, except when you need an `interface`.
  - Benefit: using the simpler concept with terser syntax in most places.
- Use `interface` for everything.
  - Benefit: less to think about, especially if the linter reminds us when we forget.

On the balance, it seems like @matuzalemsteles policy ("usually always use `interface`") is probably the "best", although I suspect it doesn't matter at a cosmological scale. 😂